analyses/interaction-plots/scripts/01-process_mutations.R

```diff
@@ -0,0 +1,209 @@
+#
+# JA Shapiro for ALSF - CCDL
+#
+# 2019
+#
+# Generates a table of gene by gene co-occurence data with p values from Fisher's exact test.
+# By default, performs analysis of the top 50 most mutated genes.
+#
+# Option descriptions
+#
+# --maf :  Relative file path to MAF file to be analyzed. Can be .gz compressed.
+#       File path is given from top directory of 'OpenPBTA-analysis'.
+# --metadata : Relative file path to MAF file to be analyzed. Can be .gz compressed.
+#       File path is given from top directory of 'OpenPBTA-analysis'.
+# --specimen_list: A file of specimens to include. Ideally, this list should consist
+#       of independent samples (at most one from each individual). File path is given
+#       from the top directory of 'OpenPBTA-analysis'.
+# --vaf: Minimum variant allele fraction of mutations to include.
+# --n_genes: number of genes to plot interation data for (uses the most mutated n genes)
+# --out: Output file location
+#                 
+#
+# Command line example:
+#
+# Rscript analyses/interaction-plots/01-process-mutations.R \
+#   --maf data/pbta-snv-lancet.vep.maf.gz
+#   --metadata data/pbta-histologies.tsv 
+#   --specimen_list analysis/independent-samples/results/independent-specimens.wgs.primary.tsv
+
+#### Initial Set Up
+# Establish base dir
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+script_root <- file.path(root_dir, "analyses", "interaction-plots", "scripts")
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Load libraries:
+library(optparse)
+library(ggplot2)
+
+
+# Load functions
+source(file.path(script_root, "cooccur_functions.R"))
+
+#### Set up options 
+option_list <- list(
+  make_option(
+    opt_str = "--maf", type = "character", 
+    default = file.path("data", "pbta-snv-lancet.vep.maf.gz"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis') 
+              to MAF file to be analyzed. Can be .gz compressed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--seg", type = "character", 
+    default = file.path("data", "pbta-cnv-cnvkit.seg.gz"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis') 
+            to CNV file to be analyzed in seg format. Can be .gz compressed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--metadata", type = "character", 
+    default = file.path("data", "pbta-histologies.tsv"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis')
+            to MAF file to be analyzed. Can be .gz compressed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--out", type = "character", 
+    default = file.path("analyses", "interaction-plots", "results", "cooccurence.tsv"),
+    help = "Relative file path (from top directory of 'OpenPBTA-analysis')
+            where output table will be placed.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--specimen_list", type = "character", 
```

Comment:
Are you planning on leveraging this option to filter to disease types or would you split up the MAF data.frames based on something in the metadata? This is not important yet, but trying to understand where this is headed.

Reply:
I hadn't yet decided on this. I think maybe generating the specimen lists in a separate script, then using them here with little change may be the way to go. Different types may also have different gene lists of interest, but I haven't plugged that in as a script level option yet. The main calculation function can handle arbitrary gene lists though, so it should be easy to add that.